### PR TITLE
docs(cloud_firestore_odm): Update documentation with a brief outline of what an "odm" does.

### PR DESCRIPTION
**Pull request by KikiManjaro**
## Description

First contribution ever
[📚] Add what "ODM" stands for #7780
Add a paragraph for what ODM stands for in [here](https://firebase.flutter.dev/docs/firestore-odm/overview/)

## Related Issues

Closes https://github.com/FirebaseExtended/flutterfire/issues/7780

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/firestore-odm/overview.mdx [hunk 1]
```diff
@@ -10,6 +10,12 @@ See the [discussion](https://github.com/FirebaseExtended/flutterfire/discussions
 
 :::
 
+## What is an ODM ?
+
+The Object Document Model or ODM is an interface that treats a document as a tree structure wherein each node is an object representing a part of the document. .
```
Comment (rrousselGit) [inline docs/firestore-odm/overview.mdx:15]:
```suggestion
The Object Document Mapper or ODM is an interface that treats a document as a tree structure wherein each node is an object representing a part of the document.   
```

**Comment by KikiManjaro** [comment]
I don't understand why the task `website / preview (pull_request) Failing after 1m` is failling. Does someone have an idea ?

File: docs/firestore-odm/overview.mdx [hunk 2]
```diff
@@ -10,6 +10,12 @@ See the [discussion](https://github.com/FirebaseExtended/flutterfire/discussions
 
 :::
 
+## What is an ODM ?
+
+The Object Document Mapper or ODM is an interface that treats a document as a tree structure wherein each node is an object representing a part of the document.   
+Each branch of the tree ends in a node, and each node contains objects. 
+ODM methods allow programmatic access to the tree; with them one can change the structure, style or content of a document.
```
Comment (russellwheatley) [inline docs/firestore-odm/overview.mdx:17]:
```suggestion
ODM methods allow programmatic access to the tree making it possible to change the structure, style or content of a document. 
```

File: docs/firestore-odm/overview.mdx [hunk 3]
```diff
@@ -10,6 +10,12 @@ See the [discussion](https://github.com/FirebaseExtended/flutterfire/discussions
 
 :::
 
+## What is an ODM ?
```
Comment (russellwheatley) [inline docs/firestore-odm/overview.mdx:13]:
```suggestion
## What is an ODM?
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Small tweaks.

File: docs/firestore-odm/overview.mdx (hunk 3 above)
Comment (russellwheatley) [reply docs/firestore-odm/overview.mdx:13]: @KikiManjaro thanks for the PR, some small tweaks 😄

**Comment by rrousselGit** [comment]
I think this is good to go. Let's merge it

**Pull request merged**